doc/source/train/user-guides/experiment-tracking.rst

```diff
@@ -1,195 +1,302 @@
 
-.. tab-set::
+Step 1: Choose a tracking service to use
+----------------------------------------
 
-    .. tab-item:: PyTorch
+The following tracking service setups have the simplest configurations for remote nodes
+to access and write logs to.
 
-        You can use experiment tracking libraries such as Weights & Biases, Mlflow, or
-        Comet directly in your Ray Train training loop.
+**W&B:**
 
-        There are two things to keep in mind:
+- online mode: Should work out-of-the-box with correct credentials
 
-        1. Your code is executed in parallel on many workers. However, you often only want to report
-        results from one of these workers (usually the first worker - the "rank 0" worker).
+- offline mode: Set Wandb directory to point to a shared storage path: :code:`wandb.init(dir="some_shared_storage_path/wandb")`      
 
-        2. When using the native libraries, you should report the results to both Ray Train and
-        the experiment tracking library.
+**MLflow:**
 
-        Example:
+- online mode: Start the run with externally hosted MLflow by Databricks: :code:`mlflow.start_run(tracking_uri="databricks")`
 
-        .. code-block:: python
+- offline mode: Start the run by setting tracking uri to a shared storage path: :code:`mlflow.start_run(tracking_uri="file:some_shared_storage_path/mlruns")`
+
+**TensorBoard:**
+
+- Set up ``SummaryWriter`` to write to a shared storage path: :code:`writer = SummaryWriter("some_shared_storage_path/runs")`
+
+Step 2: Make sure that all nodes and processes have access to credentials
+-------------------------------------------------------------------------
+
+**W&B:**
+
+- :code:`wandb.login(key="your_api_key")` or :code:`os.environ["WANDB_API_KEY"] = "your_api_key"`
+
+**MLflow:**
+
+- This is done by having ``databrickscfg`` accessible by all nodes. See below for code snippet.
+
+Step 3: Initialize the run 
+--------------------------
+
+Ray Train provides a training context that provides access to training identifiers. For example, 
+
+* Training ID (:meth:`context.get_trial_id() <ray.train.context.TrainContext.get_trial_id>`) 
+* Training Name (:meth:`context.get_trial_name() <ray.train.context.TrainContext.get_trial_name>`)
+
+Step 4: Log
+-----------
+
+For some logs, every Worker generates an identical copy and saving a single copy is sufficient. 
+Ray Train can save logs from the rank 0 Worker only with the following method:
+:meth:`context.get_world_rank() <ray.train.context.TrainContext.get_world_rank>`.
+
+Step 5: Finish the run
+----------------------
 
+For frameworks that require a call to mark a run as finished, include the appropriate call.
+For example, ``wandb.finish()``.
+
+Code Example
+============
+
+Let's see how the above works with some code.
+
+The following session uses Wandb and MLflow but it is adaptable to other frameworks.
+
+Pytorch
+-------
+
+Conceptual code snippets
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. tabs::
+
+    .. tab:: Wandb(online)
```

Comment:
What do you think about two layers of tabs so that we can keep "Weights & Biases" as a group-tab?